.github/workflows/build.yaml

```diff
@@ -0,0 +1,21 @@
+name: Test & Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test-unit:
+    name: Unit Testing
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Run Unit Tests Docker
+        run: make api-unit-test
```

Comment:
This should probably be run inside `main.yml` rather than separately, so that if tests fail it will block the rest of the pipeline

Reply:
Ah right sure :)